sklearn/kernel_ridge/kernel_ridge.py

```diff
@@ -143,23 +114,42 @@ def fit(self, X, y=None, sample_weight=None):
         self : returns an instance of self.
         """
         # Convert data
-        X, y = check_X_y(X, y, accept_sparse=("csr", "csc"), multi_output=True,
-                         y_numeric=True)
         if sample_weight is not None and not isinstance(sample_weight, float):
             sample_weight = check_array(sample_weight, ensure_2d=False)
 
         K = self._get_kernel(X)
         alpha = np.atleast_1d(self.alpha)
 
+        copy = self.kernel == "precomputed"
+
         ravel = False
-        if len(y.shape) == 1:
-            y = y.reshape(-1, 1)
-            ravel = True
+        if self._estimator_type == "regressor":
+            X, y_ = check_X_y(X, y, accept_sparse=("csr", "csc"),
+                              multi_output=True, y_numeric=True)
+            if len(y_.shape) == 1:
+                y_ = y_.reshape(-1, 1)
+                ravel = True
+        elif self._estimator_type == "classifier":  # Multiclass classification
+            X, y = check_X_y(X, y)
+            # Check alpha
+            if self.alpha <= 0:
+                raise ValueError('alpha must be positive')
+
+            # y is going to be encoded by zero arrays
```

Comment:
I think this is unnecessary, in the sense that the reader should be aware of what LabelBinarizer is for, or can find out

Reply:
Comments removed